public/app/features/dashboard/dashgrid/DashboardLibrary/TemplateDashboardModal.tsx

```diff
@@ -0,0 +1,172 @@
+import { css } from '@emotion/css';
+import { useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom-v5-compat';
+import { useAsync } from 'react-use';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { t, Trans } from '@grafana/i18n';
+import { getBackendSrv, getDataSourceSrv, locationService } from '@grafana/runtime';
+import { Box, Grid, Modal, Text, useStyles2 } from '@grafana/ui';
+
+import { DASHBOARD_LIBRARY_ROUTES } from '../types';
+
+import { DashboardCard } from './DashboardCard';
+import {
+  DashboardLibraryInteractions,
+  EVENT_LOCATIONS,
+  SourceEntryPoint,
+  TemplateDashboardSourceEntryPoint,
+} from './interactions';
+import { GnetDashboard, GnetDashboardsResponse, Link } from './types';
+
+const SourceEntryPointMap: Record<string, SourceEntryPoint> = {
+  quickAdd: TemplateDashboardSourceEntryPoint.QUICK_ADD_BUTTON,
+  commandPalette: TemplateDashboardSourceEntryPoint.COMMAND_PALETTE,
+  createNewButton: TemplateDashboardSourceEntryPoint.BROWSE_DASHBOARDS_PAGE,
+};
+
+export const TemplateDashboardModal = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const isOpen = searchParams.get('templateDashboards') === 'true';
+  const entryPoint = searchParams.get('source') || '';
+
+  const testDataSource = getDataSourceSrv().getList({ type: 'grafana-testdata-datasource' })[0];
+
+  const styles = useStyles2(getStyles);
+
+  const onClose = () => {
+    searchParams.delete('templateDashboards');
+    setSearchParams(searchParams);
+  };
+
+  const onPreviewDashboardClick = async (dashboard: GnetDashboard) => {
+    const sourceEntryPoint = SourceEntryPointMap[entryPoint] || 'unknown';
+    DashboardLibraryInteractions.itemClicked({
+      contentKind: 'template_dashboard',
+      datasourceTypes: [String(testDataSource?.type)],
+      libraryItemId: String(dashboard.id),
+      libraryItemTitle: dashboard.name,
+      sourceEntryPoint,
+      eventLocation: EVENT_LOCATIONS.BROWSE_DASHBOARDS_PAGE,
+      discoveryMethod: 'browse',
+    });
+
+    const params = new URLSearchParams({
+      datasource: testDataSource?.uid || '',
+      title: dashboard.name,
+      pluginId: String(testDataSource?.uid) || '',
```

Comment:
If we pass the uid, we will track in the saving step something like "pluginId=PD8C576611E62080A", instead of the grafana-test-datasource